homework2.ts

```diff
@@ -0,0 +1,64 @@
+interface AreaInterface {
+    makeElement() : any;
+}
+
+class Square implements AreaInterface {
+    name: string;
+    width: number;
+  
+    constructor(name: string, width: number) {
+        this.name = name;
+        this.width = width;
+    }
+  
+    makeElement() {
+        let area: number;
+        area = this.width * this.width;
+        return "The area of " + this.name + " is " + area;
+    }
+}
+  
+class Rectangle implements AreaInterface {
+    name: string;
+    width: number;
+    height: number;
+  
+    constructor(name: string, width: number, height: number) {
+        this.name = name;
+        this.width = width;
+        this.height = height;
+    }
+  
+    makeElement() {
+        let area: number;
+        area = this.width * this.height;
+        return "The area of " + this.name + " is " + area;
+    }
+}
+class arrayForShapesArea {
+    areaList: object[] = [];
+    
+    pushToArray1(shapeArray: any[]) {
+        for (let i: number = 0; i < shapeArray.length; i++) {
+            this.pushToArray2(shapeArray[i]);
+        }
+    }
+
+    pushToArray2(areaInterfaceTypedVar : AreaInterface) {
+        this.areaList.push(areaInterfaceTypedVar.makeElement());
+    }
+}
+
+
+const Square1 = new Square("Square1", 3);
+const Rectangle1 = new Rectangle("Rectangle1", 5, 6);
+const Square2 = new Square("Square2", 8);
+const Rectangle2 = new Rectangle("Rectangle2", 10, 15);
+
+const objectList: object[] = [Square1, Rectangle1, Square2, Rectangle2];
```

Comment:
The object list should be Shape Interfaces type

Reply:
I made an example here, after you fixed this homework, let refer to the example https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgMoAs4AcIElzTxLIDeAUMpcgOYRgBycAthABQCUAXMgM5hShqAbgpUEcADYBXCXEgBBKBDgdk3EFKYAjaCIC+ZMglk8eaAI5S4S5MCZYJEFuDMZseArEQpyVZCGYIbj4BEGFRSgB3YAATMHR1TR0oEQjkBAB7EBCpBDAMqFYAlmD+QQAaZGi4hP8k6HZSNL944B4AOmKUAF5-QJE-FvQ29ur45F6x9AGqAzTxaVkFJRVG30HKRzBka2VE7V1mql24CeRWjqnkACpz4cvY+JmNpTApKBAdlefkOb9aBiBVTrF50d6fC6dfppAxzYxwUzIABKEDycDCjls9kczjArkwOHwkC8xBBfRKvDKYR+U32yR+6AgwGo6DAdMO8yyOTyBSKgVKoWolVpdQOUEqjOZrPZUDWR0okK6Zy6PyGIyuk0e03ldxGkpZ216+tZPz+YkkMjkEEUymBOq2Xz2ovpOpOZ0hV1ukONYFVVFe4MoJ1Nhn+dEYLDtG0oAY+uo6KphZDmRi521Qll2AEZlRBIhYrEpWAAiDOFiBZ4uVADM7BEmWy2xRaIxFdz+ebYHR1EcJc73cclcqAFZKgA2OupxsF3YAJnbM6LpczSlnVeQAA5Jw2+MjUV3W-PeiA83uWz22MX+4f11mAAyVLPDydT3cZLQAK33ABk2my0ASHjEkQEAANoALpnKBZbZpU14XlmlQwaucH7gOECzuBqTvl+eS-nw7QwAUACiiDoKwMBSCAeTAFkrA8IBcp+Du2xMDw1BnAABgAKoyjqnBkMDIAAJCQDHuO0AIRmw7B6LYZiieJODtAslrLLasmcZy2QZI47QSBk1CsGx1CTno7BkEAA